<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<meta name="date" content="2018-07-31" />

<title>Organoid Batch 1 Quality Control</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Combes Organoid Paper Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Organoid Batch 1 Quality Control</h1>
<h4 class="date"><em>2018-07-31</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-07-31</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20180730)</code> </summary></p>
<p>The command <code>set.seed(20180730)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="http://bioinformatics.pages.mcri.edu.au/combes-organoid-paper//tree/74d5c23962d854d8be3fd78799e8e15824d843f7" target="_blank">74d5c23</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/cache/
    Ignored:    packrat/lib-R/
    Ignored:    packrat/lib-ext/
    Ignored:    packrat/lib/
    Ignored:    packrat/src/
    Ignored:    public/figure/

Untracked files:
    Untracked:  01_Organoid123_QC.Rmd/

Unstaged changes:
    Modified:   analysis/01_Organoid123_QC.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/http://bioinformatics.pages.mcri.edu.au/combes-organoid-paper//4536ef665ceb0e6f48bd38173dfb2e27a7a41ceb/public/01_Organoid123_QC.html" target="_blank">4536ef6</a>
</td>
<td style="text-align:left;">
Luke Zappia
</td>
<td style="text-align:left;">
2018-07-31
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="http://bioinformatics.pages.mcri.edu.au/combes-organoid-paper//blob/bb63ce238cb5e1b29aa7d090b55e9d24c84b9fc4/analysis/01_Organoid123_QC.Rmd" target="_blank">bb63ce2</a>
</td>
<td style="text-align:left;">
Luke Zappia
</td>
<td style="text-align:left;">
2018-07-31
</td>
<td style="text-align:left;">
Add organoids batch 1 QC
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this document we are going to read in the first batch of our Organoid data, produce various quality control plots and remove any low-quality cells or uninformative genes.</p>
</div>
<div id="qc-plots" class="section level1">
<h1>QC plots</h1>
<div id="by-cell" class="section level2">
<h2>By cell</h2>
<div id="total-counts" class="section level3">
<h3>Total counts</h3>
<p>Violin plots of the library size (total counts) for each of the samples.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/total-counts-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="number-of-features-by-library-size" class="section level3">
<h3>Number of features by library size</h3>
<p>Relationship between the total counts for each cell and the number of expressed genes. We expect the number of genes to increase with the number of counts, hopefully reaching saturation.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/count-features-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="pca" class="section level3">
<h3>PCA</h3>
<p>PCA plots coloured by different variables.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/pca-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="t-sne" class="section level3">
<h3>t-SNE</h3>
<p>t-SNE plots coloured by different variables.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/t-SNE-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="explantory-variables" class="section level3">
<h3>Explantory variables</h3>
<p>Plots of the variance explained by various variables.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/exp-vars-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Correlation between explanatory variables.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/exp-vars-pairs-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="rle" class="section level3">
<h3>RLE</h3>
<p>Relative Log Expression (RLE) plots. Ideally all boxes should be aligned and have the size size.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/rle-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="mitochondrial-genes" class="section level3">
<h3>Mitochondrial genes</h3>
<p>Looking at the effect of mitchondrial genes. We define mitochondrial genes as genes on the MT chromosome or with “mitochondrial” in the description.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/pct-mt-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="ribosomal-genes" class="section level3">
<h3>Ribosomal genes</h3>
<p>Looking at the effect of ribosomal genes. We define ribosomal genes as genes with “ribosom” in the description.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/pct-ribo-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="housekeeping-genes" class="section level3">
<h3>Housekeeping genes</h3>
<p>Plots of housekeeping genes. We may want to use these for filtering as a proxy for the health of the cell.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/hk-exprs-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="by-gene" class="section level2">
<h2>By gene</h2>
<div id="high-expression" class="section level3">
<h3>High expression</h3>
<p><img src="figure/01_Organoid123_QC.Rmd/high-expression-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="expression-frequency-by-mean" class="section level3">
<h3>Expression frequency by mean</h3>
<p><img src="figure/01_Organoid123_QC.Rmd/mean-expression-freq-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="total-counts-by-num-cells-expressed" class="section level3">
<h3>Total counts by num cells expressed</h3>
<p><img src="figure/01_Organoid123_QC.Rmd/gene-expression-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="filtering" class="section level1">
<h1>Filtering</h1>
<p>To begin with we have 7004 cells across with 33694 features from the ENSEMBL annotation.</p>
<div id="cells" class="section level2">
<h2>Cells</h2>
<div id="quantification" class="section level3">
<h3>Quantification</h3>
<p>Let’s consider how many reads are assigned to features. We can plot the total number of counts in each cell against the number of genes that are expressed.</p>
<p>Cells that have been filtered are shown as triangles.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/quantification-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Cells that express many genes are potential multiplets (multiple cells captured in a single droplet). We will remove 11 cells with more than 8000 genes expressed.</p>
<p>We now have 6993 cells.</p>
</div>
<div id="mitochondrial-genes-1" class="section level3">
<h3>Mitochondrial genes</h3>
<p>Over-expression of mitochondrial genes can be an indication that a cell is stressed or damaged in some way. Let’s have a look at the percentage of counts that are assigned to mitchondrial genes.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/mt-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Some of the cells show high proportions of MT counts. We will remove 118 cells with greater than 10% MT counts.</p>
<p>That leaves 6876 cells.</p>
</div>
<div id="ribsomal-genes" class="section level3">
<h3>Ribsomal genes</h3>
<p>We can do a similar thing for ribosomal gene expression.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/ribosomal-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Some of the cells show high proportions of ribosomal counts. We will remove 183 cells with greater than 45% ribosomal counts.</p>
<p>That leaves 6693 cells.</p>
</div>
<div id="housekeeping" class="section level3">
<h3>Housekeeping</h3>
<p>Similarly we can look at the expression of the “housekeeping” genes GAPDH and ACTB.</p>
<p><img src="figure/01_Organoid123_QC.Rmd/housekeeping-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>We will remove cells where ACTB is expressed below 2 or GAPDH is expressed below 2. This removes 53 cells.</p>
<p>After filtering we are left with 6649 cells.</p>
</div>
<div id="dimensionality-reduction-filtered-cells" class="section level3">
<h3>Dimensionality reduction (filtered cells)</h3>
<p>Now that we are relatively confident we have a set of good quality cells, let’s see what they look like in reduced dimensions.</p>
<div id="pca-1" class="section level4">
<h4>PCA</h4>
<p><img src="figure/01_Organoid123_QC.Rmd/pca-filtered-cells-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="t-sne-1" class="section level4">
<h4>t-SNE</h4>
<p><img src="figure/01_Organoid123_QC.Rmd/tSNE-filtered-cells-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="genes" class="section level2">
<h2>Genes</h2>
<div id="expression" class="section level3">
<h3>Expression</h3>
<p>Some of the features we would never expect to see expressed in an RNA-seq experiment. Before doing anything else I am going to remove the features that have less than two counts across all cells.</p>
<p>This removes 10784 genes and leaves us with 22910.</p>
<p>We will also going remove genes that are expressed in less than two individual cells.</p>
<p>This removes 35 genes and leaves us with 22875.</p>
</div>
<div id="hgnc-genes" class="section level3">
<h3>HGNC genes</h3>
<p>We are also going to filter out any genes that don’t have HGNC symbols. These are mostly pseudogenes and are unlikely to be informative.</p>
<p>This removes 4489 genes and leaves us with 18386.</p>
<p>There are 1 gene(s) with duplicate HGNC symbol names. For these genes we will use an alternative symbol name. Once we have done this we can rename the features using feature symbol inseated of ENSEMBL IDs which will make interpreting results easier.</p>
</div>
<div id="dimensionality-reduction-filtered-genes" class="section level3">
<h3>Dimensionality reduction (filtered genes)</h3>
<p>Let’s see what our final dataset looks like in reduced dimensions.</p>
<div id="pca-2" class="section level4">
<h4>PCA</h4>
<p><img src="figure/01_Organoid123_QC.Rmd/pca-filtered-genes-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="t-sne-2" class="section level4">
<h4>t-SNE</h4>
<p><img src="figure/01_Organoid123_QC.Rmd/tSNE-filtered-genes-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>We now have a high-quality dataset for our analysis with 18386 genes and 6649 cells. A median of 2738 genes are expressed in each cell.</p>
</div>
</div>
</div>
</div>
<div id="metadata" class="section level1">
<h1>Metadata</h1>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre><code> setting  value                       
 version  R version 3.5.0 (2018-04-23)
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 tz       Australia/Melbourne         
 date     2018-07-31                  

 package              * version   date      
 AnnotationDbi          1.42.1    2018-07-30
 assertthat             0.2.0     2017-04-11
 backports              1.1.2     2017-12-13
 base                 * 3.5.0     2018-06-18
 beeswarm               0.2.3     2016-04-25
 bindr                  0.1.1     2018-03-13
 bindrcpp             * 0.2.2     2018-03-29
 Biobase              * 2.40.0    2018-07-30
 BiocGenerics         * 0.26.0    2018-07-30
 BiocParallel         * 1.14.2    2018-07-08
 biomaRt                2.36.1    2018-07-30
 bit                    1.1-14    2018-05-29
 bit64                  0.9-7     2017-05-08
 bitops                 1.0-6     2013-08-17
 blob                   1.1.1     2018-03-25
 broom                  0.5.0     2018-07-17
 cellranger             1.1.0     2016-07-27
 cli                    1.0.0     2017-11-05
 codetools              0.2-15    2016-10-05
 colorspace             1.3-2     2016-12-14
 compiler               3.5.0     2018-06-18
 cowplot              * 0.9.3     2018-07-15
 crayon                 1.3.4     2017-09-16
 curl                   3.2       2018-03-28
 data.table             1.11.4    2018-05-27
 datasets             * 3.5.0     2018-06-18
 DBI                    1.0.0     2018-05-02
 DelayedArray         * 0.6.2     2018-07-23
 DelayedMatrixStats     1.2.0     2018-07-30
 devtools               1.13.6    2018-06-27
 digest                 0.6.15    2018-01-28
 dplyr                * 0.7.6     2018-06-29
 DropletUtils           1.0.2     2018-07-30
 DT                     0.4       2018-01-30
 dynamicTreeCut         1.63-1    2016-03-11
 edgeR                  3.22.3    2018-06-21
 evaluate               0.10.1    2017-06-24
 FNN                    1.1       2013-07-31
 forcats              * 0.3.0     2018-02-19
 GenomeInfoDb         * 1.16.0    2018-07-30
 GenomeInfoDbData       1.1.0     2018-07-30
 GenomicRanges        * 1.32.6    2018-07-20
 ggbeeswarm             0.6.0     2017-08-07
 ggplot2              * 3.0.0     2018-07-03
 git2r                  0.21.0    2018-01-04
 glue                   1.3.0     2018-07-17
 graphics             * 3.5.0     2018-06-18
 grDevices            * 3.5.0     2018-06-18
 grid                   3.5.0     2018-06-18
 gridExtra              2.3       2017-09-09
 gtable                 0.2.0     2016-02-26
 haven                  1.1.2     2018-06-27
 here                 * 0.1       2017-05-28
 hms                    0.4.2     2018-03-10
 htmltools              0.3.6     2017-04-28
 htmlwidgets            1.2       2018-04-19
 httpuv                 1.4.5     2018-07-19
 httr                   1.3.1     2017-08-20
 igraph                 1.2.2     2018-07-27
 IRanges              * 2.14.10   2018-07-30
 jsonlite               1.5       2017-06-01
 knitr                  1.20      2018-02-20
 labeling               0.3       2014-08-23
 later                  0.7.3     2018-06-08
 lattice                0.20-35   2017-03-25
 lazyeval               0.2.1     2017-10-29
 limma                  3.36.2    2018-06-21
 locfit                 1.5-9.1   2013-04-20
 lubridate              1.7.4     2018-04-11
 magrittr               1.5       2014-11-22
 Matrix               * 1.2-14    2018-04-09
 matrixStats          * 0.54.0    2018-07-23
 memoise                1.1.0     2017-04-21
 methods              * 3.5.0     2018-06-18
 mgcv                   1.8-24    2018-06-18
 mime                   0.5       2016-07-07
 modelr                 0.1.2     2018-05-11
 munsell                0.5.0     2018-06-12
 nlme                   3.1-137   2018-04-07
 parallel             * 3.5.0     2018-06-18
 pillar                 1.3.0     2018-07-14
 pkgconfig              2.0.1     2017-03-21
 plyr                   1.8.4     2016-06-08
 prettyunits            1.0.2     2015-07-13
 progress               1.2.0     2018-06-14
 promises               1.0.1     2018-04-13
 purrr                * 0.2.5     2018-05-29
 R.methodsS3            1.7.1     2016-02-16
 R.oo                   1.22.0    2018-04-22
 R.utils                2.6.0     2017-11-05
 R6                     2.2.2     2017-06-17
 Rcpp                   0.12.18   2018-07-23
 RCurl                  1.95-4.11 2018-07-15
 readr                * 1.1.1     2017-05-16
 readxl                 1.1.0     2018-04-20
 reshape2               1.4.3     2017-12-11
 rhdf5                  2.24.0    2018-07-30
 Rhdf5lib               1.2.1     2018-07-30
 rjson                  0.2.20    2018-06-08
 rlang                  0.2.1     2018-05-30
 rmarkdown              1.10.2    2018-07-30
 rprojroot              1.3-2     2018-01-03
 RSQLite                2.1.1     2018-05-06
 rstudioapi             0.7       2017-09-07
 Rtsne                  0.13      2017-04-14
 rvest                  0.3.2     2016-06-17
 S4Vectors            * 0.18.3    2018-07-30
 scales                 0.5.0     2017-08-24
 scater               * 1.8.2     2018-07-27
 scran                  1.8.2     2018-07-30
 shiny                  1.1.0     2018-05-17
 shinydashboard         0.7.0     2018-03-21
 SingleCellExperiment * 1.2.0     2018-07-30
 statmod                1.4.30    2017-06-18
 stats                * 3.5.0     2018-06-18
 stats4               * 3.5.0     2018-06-18
 stringi                1.2.4     2018-07-20
 stringr              * 1.3.1     2018-05-10
 SummarizedExperiment * 1.10.1    2018-07-30
 tibble               * 1.4.2     2018-01-22
 tidyr                * 0.8.1     2018-05-18
 tidyselect             0.2.4     2018-02-26
 tidyverse            * 1.2.1     2017-11-14
 tools                  3.5.0     2018-06-18
 tximport               1.8.0     2018-07-30
 utils                * 3.5.0     2018-06-18
 vipor                  0.4.5     2017-03-22
 viridis                0.5.1     2018-03-29
 viridisLite            0.3.0     2018-02-01
 whisker                0.3-2     2013-04-28
 withr                  2.1.2     2018-03-15
 workflowr              1.1.1     2018-07-06
 XML                    3.98-1.12 2018-07-15
 xml2                   1.2.0     2018-01-24
 xtable                 1.8-2     2016-02-05
 XVector                0.20.0    2018-07-30
 yaml                   2.2.0     2018-07-25
 zlibbioc               1.26.0    2018-07-30
 source                            
 Bioconductor                      
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 local                             
 CRAN (R 3.5.0)                    
 cran (@0.1.1)                     
 cran (@0.2.2)                     
 Bioconductor                      
 Bioconductor                      
 Bioconductor                      
 Bioconductor                      
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 cran (@1.0-6)                     
 CRAN (R 3.5.0)                    
 cran (@0.5.0)                     
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 cran (@1.3-2)                     
 local                             
 cran (@0.9.3)                     
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 cran (@1.11.4)                    
 local                             
 CRAN (R 3.5.0)                    
 Bioconductor                      
 Bioconductor                      
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 cran (@0.7.6)                     
 Bioconductor                      
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 Bioconductor                      
 CRAN (R 3.5.0)                    
 cran (@1.1)                       
 CRAN (R 3.5.0)                    
 Bioconductor                      
 Bioconductor                      
 Bioconductor                      
 CRAN (R 3.5.0)                    
 cran (@3.0.0)                     
 CRAN (R 3.5.0)                    
 cran (@1.3.0)                     
 local                             
 local                             
 local                             
 cran (@2.3)                       
 cran (@0.2.0)                     
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 cran (@1.2)                       
 cran (@1.4.5)                     
 CRAN (R 3.5.0)                    
 cran (@1.2.2)                     
 Bioconductor                      
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 cran (@0.3)                       
 cran (@0.7.3)                     
 CRAN (R 3.5.0)                    
 cran (@0.2.1)                     
 Bioconductor                      
 CRAN (R 3.5.0)                    
 cran (@1.7.4)                     
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 local                             
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 cran (@0.5.0)                     
 CRAN (R 3.5.0)                    
 local                             
 cran (@1.3.0)                     
 cran (@2.0.1)                     
 cran (@1.8.4)                     
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 cran (@1.0.1)                     
 cran (@0.2.5)                     
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 cran (@0.12.18)                   
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 cran (@1.4.3)                     
 Bioconductor                      
 Bioconductor                      
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 Github (rstudio/rmarkdown@18207b9)
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 cran (@0.13)                      
 CRAN (R 3.5.0)                    
 Bioconductor                      
 cran (@0.5.0)                     
 Bioconductor                      
 Bioconductor                      
 cran (@1.1.0)                     
 CRAN (R 3.5.0)                    
 Bioconductor                      
 CRAN (R 3.5.0)                    
 local                             
 local                             
 cran (@1.2.4)                     
 CRAN (R 3.5.0)                    
 Bioconductor                      
 cran (@1.4.2)                     
 cran (@0.8.1)                     
 cran (@0.2.4)                     
 CRAN (R 3.5.0)                    
 local                             
 Bioconductor                      
 local                             
 CRAN (R 3.5.0)                    
 cran (@0.5.1)                     
 cran (@0.3.0)                     
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 CRAN (R 3.5.0)                    
 cran (@1.8-2)                     
 Bioconductor                      
 cran (@2.2.0)                     
 Bioconductor                      </code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
